profile2/tool/src/java/org/sakaiproject/profile2/tool/entityprovider/ProfileEntityProvider.java

```diff
@@ -251,7 +252,26 @@ public Object getConnectionStatus(EntityReference ref, Map<String, Object> param
 		
 		return connectionsLogic.getConnectionStatus(uuid, parameters.get("friendId").toString());
 	}
-	
+
+    @EntityCustomAction(action="unreadMessagesCount",viewKey=EntityView.VIEW_SHOW)
+	public Object getUnreadMessagesCount(EntityReference ref) {
+
+		if(!sakaiProxy.isLoggedIn()) {
+			throw new SecurityException("You must be logged in to get the unread messages count.");
+		}
+
+		//convert input to uuid
+		String uuid = sakaiProxy.ensureUuid(ref.getId());
+		if(StringUtils.isBlank(uuid)) {
+			throw new EntityNotFoundException("Invalid user.", ref.getId());
+		}
+        
+		if (!sakaiProxy.getCurrentUserId().equals(uuid)) {
```

Comment:
I didn't think admins would be that interested, but, it's easy so ...

I'll rejig it :)

On 8 March 2016 at 15:19, Code Hugger (Matthew Jones) <
notifications@github.com> wrote:

> In
> profile2/tool/src/java/org/sakaiproject/profile2/tool/entityprovider/ProfileEntityProvider.java
> <https://github.com/sakaiproject/sakai/pull/1883#discussion_r55373000>:
>
> > -	
> > +
> > +    @EntityCustomAction(action="unreadMessagesCount",viewKey=EntityView.VIEW_SHOW)
> > +	public Object getUnreadMessagesCount(EntityReference ref) {
> > +
> > +		if(!sakaiProxy.isLoggedIn()) {
> > +			throw new SecurityException("You must be logged in to get the unread messages count.");
> > +		}
> > +
> > +		//convert input to uuid
> > +		String uuid = sakaiProxy.ensureUuid(ref.getId());
> > +		if(StringUtils.isBlank(uuid)) {
> > +			throw new EntityNotFoundException("Invalid user.", ref.getId());
> > +		}
> > +
> > +		if (!sakaiProxy.getCurrentUserId().equals(uuid)) {
>
> I'd think if you're an admin user calling this, you could check any
> message count
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sakaiproject/sakai/pull/1883/files#r55373000>.
>